tests/unit-tests/dds/DCPS/ConfigStoreImpl.cpp

```diff
@@ -173,57 +184,146 @@ TEST(dds_DCPS_ConfigStoreImpl, set_get_TimeDuration_milliseconds)
+      store.set("key", value_optional_port, ConfigStoreImpl::Format_Optional_Port, ConfigStoreImpl::Kind_IPV4);
+      EXPECT_EQ(store.get("key", default_value, ConfigStoreImpl::Format_Optional_Port, ConfigStoreImpl::Kind_IPV4), value_optional_port);
+    }
+
+    {
+      store.set_string("key", "not a network address");
+      EXPECT_EQ(store.get("key", default_value, ConfigStoreImpl::Format_Optional_Port, ConfigStoreImpl::Kind_IPV4), default_value);
     }
   }
-  EXPECT_TRUE(found_it);
-  store.disconnect(reader);
-  store.set_string("key", "value2");
-  reader->take(datas, infos, DDS::LENGTH_UNLIMITED,
-               DDS::ANY_SAMPLE_STATE, DDS::ANY_VIEW_STATE, DDS::ANY_INSTANCE_STATE);
-  found_it = false;
-  for (size_t idx = 0; !found_it && idx != datas.size(); ++idx) {
-    if (datas[idx] == ConfigPair("key", "value2")) {
-      found_it = true;
+
+#if ACE_HAS_IPV6
+  {
+    const NetworkAddress default_value("::");
+    EXPECT_EQ(store.get("key6", default_value, ConfigStoreImpl::Format_No_Port, ConfigStoreImpl::Kind_IPV6), default_value);
+
+    {
+      const NetworkAddress value_no_port("::1");
+      store.set("key6", value_no_port, ConfigStoreImpl::Format_No_Port, ConfigStoreImpl::Kind_IPV6);
+      EXPECT_EQ(store.get("key6", default_value, ConfigStoreImpl::Format_No_Port, ConfigStoreImpl::Kind_IPV6), value_no_port);
+    }
+
+    {
+      const NetworkAddress value_required_port("[::1]:80");
+      store.set("key6", value_required_port, ConfigStoreImpl::Format_Required_Port, ConfigStoreImpl::Kind_IPV6);
+      EXPECT_EQ(store.get("key6", default_value, ConfigStoreImpl::Format_Required_Port, ConfigStoreImpl::Kind_IPV6), value_required_port);
+    }
+
+    {
+      const NetworkAddress value_optional_port("::1");
+      store.set("key6", value_optional_port, ConfigStoreImpl::Format_Optional_Port, ConfigStoreImpl::Kind_IPV6);
+      EXPECT_EQ(store.get("key6", default_value, ConfigStoreImpl::Format_Optional_Port, ConfigStoreImpl::Kind_IPV6), value_optional_port);
+    }
+
+    {
+      const NetworkAddress value_optional_port("[::1]:80");
+      store.set("key6", value_optional_port, ConfigStoreImpl::Format_Optional_Port, ConfigStoreImpl::Kind_IPV6);
+      EXPECT_EQ(store.get("key6", default_value, ConfigStoreImpl::Format_Optional_Port, ConfigStoreImpl::Kind_IPV6), value_optional_port);
+    }
+
+    {
+      store.set_string("key6", "not a network address");
+      EXPECT_EQ(store.get("key6", default_value, ConfigStoreImpl::Format_Optional_Port, ConfigStoreImpl::Kind_IPV6), default_value);
     }
   }
-  EXPECT_FALSE(found_it);
+#endif
+}
+
+TEST(dds_DCPS_ConfigStoreImpl, take_has_prefix)
+{
+  ConfigTopic_rch topic = make_rch<ConfigTopic>();
+  ConfigStoreImpl store1(topic);
+  ConfigReader_rch reader = make_rch<ConfigReader>(store1.datareader_qos());
+  topic->connect(reader);
+
+  store1.set("PREFIX1_key", "value");
+  EXPECT_TRUE(take_has_prefix(reader, "PREFIX1"));
+  EXPECT_FALSE(take_has_prefix(reader, "PREFIX1"));
```

Comment:
This isn't really related to this code, but reading it made me think of it. Does the config store warn if a value wasn't used? If not, would it be possible? There's different things grabbing things from the key store, so I assume that could be tracked as bool on `ConfigPair` and at some point you'd know if something was used or not. If someone misunderstood how to make a key value or mistyped one, which could happen to a user or one of us, then it seems like it would be good for some sort of feedback.

Reply:
Good question.  The InternalDataReader keeps the view state and sample state.  So, if everything accesses the config through the same reader, then those states would reflect if a value was accessed or not.  Writing the function to do the scan would be easy.  The challenge would be when to invoke it.